theano/gof/opt.py

```diff
@@ -558,6 +614,39 @@ def process_node(self, fgraph, node):
         else:
             self.nodes_seen.add(node)
 
+    def get_merged_assert_input(self, node, candidate):
+        new_inputs = []
+        for node_i, cand_i in zip(node.inputs, candidate.inputs):
+            # if node_i is assert
+            if (node_i.owner and
+                    isinstance(node_i.owner.op,
+                               theano.tensor.opt.Assert)):
+                # node_i is assert, cand_i is assert
+                if (cand_i.owner and
+                        isinstance(cand_i.owner.op,
+                                   theano.tensor.opt.Assert)):
+                    # Here two assert nodes are merged.
+                    # Step 1. Check if two conditions the same one
+                    # Step 2. Combine the two with T.and_(a, b)
+                    node_cond = node_i.owner.inputs[1]
+                    cand_cond = cand_i.owner.inputs[1]
+                    if node_cond.owner is cand_cond.owner:
+                        new_inputs.append(cand_i)
+                    else:
+                        new_inputs.append(
+                            theano.tensor.opt.assert_op(
+                                node_i.owner.inputs[0],
+                                theano.tensor.and_(node_cond, cand_cond)))
```

Comment:
Should the order of conditions be kept? I use `new_cond = list(set(node_i.owner.inputs[1:] + cand_i.owner.inputs[1:]))`, so the order can be changed in the set.